### ISSUE TITLE
Per-iteration scheduler steps past T_max on webdataset loaders

### ISSUE THREAD

**Issue by timoverse** [pwhofman/probly#557]
get_scheduler` derives `total_iters` for `warmup_cosine` from `len(train_loader)`
(`src/probly_benchmark/train.py`, `_make_warmup_cosine` / `_training_loop`). For the
ImageNet webdataset loader, `len()` is the nominal `num_samples // batch_size`
(e.g. 1,281,167 // 2048 = 625), but the actual number of batches per epoch is higher:
`wds.WebDataset` splits shards across workers (`split_by_worker` is the default in
webdataset 1.0.2), and each worker's stream is batched independently, so each of the
up-to-16 workers emits its own partial final batch.

Because `_training_loop` steps the per-iteration scheduler once per *actual* batch,
`SequentialLR`/`CosineAnnealingLR` is stepped past `T_max` near the end of training.
The cosine schedule is periodic, so the learning rate **climbs back up from
`min_lr`** during the surplus steps of the final epochs instead of staying at
`min_lr`.

**Affected:** every ImageNet run using `warmup_cosine` (i.e. all methods on the
resnet50_imagenet recipe). CIFAR-10 is unaffected (`len()` is exact) and epoch-level
schedulers are unaffected.

**Proposed fix** (validated locally, then reverted to keep rebuttal results
comparable): record the budget in `_make_warmup_cosine`
(`scheduler._total_iters = total_iters`) and guard the per-iteration step in both
training loops:

```python
step_budget = getattr(scheduler, "_total_iters", None)
...
if scheduler is not None and step_per_iter and (step_budget is None or scheduler.last_epoch < step_budget):
    scheduler.step()
```

Verified behavior with the fix: after `total_iters` steps the lr stays pinned at
`min_lr` for the surplus batches.
